File: javascript/june2021/35_perimeterSum.js
// Challenge link: https://www.codewars.com/kata/559a28007caad2ac4e000083/train/javascript

function perimeter(n) {
  return 4 * fibSum(n);
}
const fibSum = (num) => {
  let fib0 = 0;
  let fib1 = 1;
  let sum = 0;
  for (let i = 0; i <= num; i++) {
    sum += fib1;
    fib1 += fib0;
    fib0 = fib1 - fib0;
  }
  return sum;
};

File: dart/35_perimeterSum.dart
// Challenge link: https://www.codewars.com/kata/559a28007caad2ac4e000083

BigInt perimeter(int m) {
  return fibSum(m) * BigInt.from(4);
}

BigInt fibSum(int n) {
  BigInt fib0 = BigInt.zero;
  BigInt fib1 = BigInt.one;
  BigInt sum = BigInt.zero;
  for (int i = 0; i <= n; i++) {
    sum += fib1;
    fib1 += fib0;
    fib0 = fib1 - fib0;
  }
  return sum;
}
